chain/client/src/client_actor.rs

```diff
@@ -671,6 +673,50 @@ impl Handler<Status> for ClientActor {
                 earliest_block_time = Some(earliest_block.timestamp());
             }
         }
+        // Provide more detailed information about the current state of chain.
+        // For now - provide info about last 50 blocks.
+        let detailed_debug_status = if msg.get_detailed_info {
+            let mut blocks_debug: Vec<DebugBlockStatus> = Vec::new();
+
+            let mut last_block_hash = head.last_block_hash;
+            let mut last_block_timestamp: u64 = 0;
+
+            // Fetch last 50 blocks (we can fetch more blocks in the future if needed)
+            for _ in 0..50 {
+                let block = match self.client.chain.get_block(&last_block_hash) {
+                    Ok(block) => block,
+                    Err(_) => break,
+                };
+                let chunks = block
+                    .chunks()
+                    .iter()
+                    .map(|chunk| DebugChunkStatus {
+                        shard_id: chunk.shard_id(),
+                        chunk_hash: chunk.chunk_hash(),
+                        gas_used: chunk.gas_used(),
+                        processing_time_ms: CryptoHashTimer::get_timer_value(chunk.chunk_hash().0),
```

Comment:
This feels potentially abusable -- here we lock that global mutex 50 times, which might contend with the actual processing work.

Reply:
yes - that's why I wanted this to be disabled by default (and enabled only on some selected canary nodes that we own)